HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+// Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+// **Пример:**
+
+// Предприятие 1 (45 сотрудников)
+// - Отдел тестирования (10 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Администрация (15 человек)
+// Предприятие 2 (75 сотрудников)
+// - Отдел разработки (50 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Отдел охраны труда (5 сотрудников)
+// Предприятие 3 (нет сотрудников)
+// - Отдел аналитики (нет сотрудников)
+
+const enterpriseStructure = (enterprises) => {
+    enterprises.map(enterprise => {
+        const totalEmployees = enterprise.departments.reduce((acc, department) => acc + department.employees_count, 0);
+        if (/.*1$/.test(totalEmployees) && !/.*11$/.test(totalEmployees)) {
+            console.log(`${enterprise.name} (${totalEmployees} сотрудник)`);
+        } else {
+        console.log(`${enterprise.name} (${totalEmployees} сотрудников)`);
+        }
+        enterprise.departments.map(department => {
+            if (/.*1$/.test(department.employees_count) && !/.*11$/.test(department.employees_count)) {
+                console.log(`- ${department.name} (${department.employees_count} сотрудник)`);
+            } else {
+            console.log(`- ${department.name} (${department.employees_count} сотрудников)`);
+            }
+        });
+    });
+}
+
+enterpriseStructure(enterprises);
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+// Функция-помощник для получения предприятия по id отдела или названию отдела:
+const getEnterpriseByDepartment = (idOrName) => enterprises.filter((enterprise) => enterprise.departments.find((department) => department.id === idOrName || department.name === idOrName));
+
+const getEnterpriseName = (idOrName) => {
+  const enterpriseFound = getEnterpriseByDepartment(idOrName);
+  if (enterpriseFound == 0) throw new Error('Нет предприятия с данным id отдела или названием отдела');
+  
+  return enterpriseFound.map(enterprise => enterprise.name).join(', ');
+}
+
+console.log(getEnterpriseName(4));
+console.log(getEnterpriseName("Отдел маркетинга"));
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+// Функция-помощник для создания нового id
+const newId = () => {
+  ids = [];
+  enterprises.forEach((enterprise) => {
+    ids.push(enterprise.id);
+    ids.push(enterprise.departments.map(department => department.id));
+  });
+  const maxId = Math.max(...ids.flat());
+  
+  return maxId + 1;
+};
+
+const addEnterprise = (newEnterprise) => {
+  enterprises.push({
+    id: newId(enterprises),
+    name: newEnterprise,
+    departments: [],
+  })
+  
+  return JSON.stringify(enterprises);
```

Comment:
```suggestion
  return enterprises;
```
Don't stringify, you're not sending this data to API)

Reply:
Thanks, got it!